File: readability.py
string = input("Text: ")
W = 0
L = 0
S = 0
for i in range(len(string)):
    if (string[i].isalpha()):
        L += 1
    elif (string[i].ispace()):
        W += 1
        while(string[i].ispace()):
            i += 1
        i -= 1
    elif (string[i] == '!' or string[i] == '?' or string[i] == '.'):
        S += 1
W += 1
z = round(0.0588 * L / W * 100 - 0.296 * S / W * 100 - 15.8)
if (z > 16):
    print("Grade 16+")
elif (z < 1):
    print("Before Grade 1")
else:
    print("Grade" + str(z))

File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    int L = 0, S = 0, W = 0, i;
    string text;
    text = get_string("Text: ");
    for (i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            L++;
        }
        else if (isspace(text[i]))
        {
            W++;
            while (isspace(text[i]))
            {
                i++;
            }
            i--;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            S++;
        }
    }
    W++;
    int z = round(0.0588 * L / W * 100 - 0.296 * S / W * 100 - 15.8) > 16 ? 17 : round(0.0588 * L / W * 100 - 0.296 * S / W * 100 -
            15.8);
    if (z == 17)
    {
        printf("Grade 16+\n");
    }
    else if (z >= 1)
    {
        printf("Grade %d\n", z);
    }
    else
    {
        printf("Before Grade 1\n");
    }
    return 0;
}

